sql/src/main/java/io/crate/expression/operator/any/AnyLikeOperator.java

```diff
@@ -38,44 +39,31 @@
 import io.crate.types.ObjectType;
 
 import java.util.List;
-import java.util.function.BiPredicate;
-import java.util.regex.Pattern;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static io.crate.expression.operator.any.AnyOperators.collectionValueToIterable;
 
-public final class AnyLikeOperator extends Operator<Object> {
+public final class AnyLikeOperator extends LikeOperator {
```

Comment:
Afaik this operator does not share much with `LikeOperator` or the methods they share a static anyway and can be used from this class as well. So I do not see a real reason the create a class hierarchy here and I'd avoid it if not necessary.  This would also avoid the String cast inside `LikeOperator`